bindings/python/pymongoarrow/types.py

```diff
@@ -75,7 +75,10 @@ def _is_decimal128_str(obj):
     Int64: lambda _: int64(),
     float: lambda _: float64(),
     int: lambda _: int64(),
-    datetime: lambda _: timestamp("ms"),  # TODO: add tzinfo support
+    # Note: we cannot infer a timezone form a raw datetime class,
+    # if a timezone is preferred then a timestamp with tz information
+    # must be used directly.
+    datetime: lambda _: timestamp("ms"),
```

Comment:
What is this used for? If we have a collection with tz_aware=True, that collection will produce UTC timezone aware datetimes. If we call find_arrow_all with the same collection, I would expect the arrow datetimes to also be UTC timezone aware. Is that not the case? Perhaps the schema discovery needs to take the collection's CodecOptions into account?

Reply:
The schema discovery does take into account the codec options.   This particular code path is only hit when the user explicitly uses the `datetime` class as the type in a schema.

Reply:
Ooooooh okay. I guess that makes sense to me. Seems like a bit of a pitfall but fair enough.

Reply:
We could return a special type here that signifies "tz unknown" and then finalize it when we create the context.

Reply:
I think the current behavior is okay for now. Could you create a ticket for that idea though? I have some more thoughts but don't want to hold up this PR.

Reply:
https://jira.mongodb.org/browse/ARROW-132